src/numeric_index.c

```diff
@@ -154,16 +154,7 @@ NRN_AddRv NumericRangeNode_Add(NumericRangeNode *n, t_docId docId, double value)
 
     if (rv.changed) {
       // if there was a split it means our max depth has increased.
-      // we are too deep - we don't retain this node's range anymore.
-      // this keeps memory footprint in check
-      if (++n->maxDepth > NR_MAX_DEPTH && n->range) {
```

Comment:
Why not keep it and just change NR_MAX_DEPTH? In case we want to return it? Maybe even make it configurable (without document) so we can easily revert it if needed?

Reply:
@MeirShpilraien - NR_MAX_DEPTH is not only used for removing ranges from parents but to balance the tree in case one child is deeper than the other. Changing in the way you describe broke the tree and would require changes to the data structure which we would like to avoid.

Reply:
Split macro to NR_MAX_DEPTH_RANGE and NR_MAX_DEPTH_BALANCE